h2o-automl/src/main/java/ai/h2o/automl/targetencoding/BroadcastJoinForTargetEncoder.java

```diff
@@ -0,0 +1,139 @@
+package ai.h2o.automl.targetencoding;
+
+import water.MRTask;
+import water.fvec.Chunk;
+import water.fvec.Frame;
+import water.util.IcedHashMap;
+
+class BroadcastJoinForTargetEncoder {
+  
+  static class CompositeLookupKey {
+    // Note: Consider using indexes of levels instead of strings to save space. It will probably require 
+    // to use CategoricalWrappedVec.computeMap() since indexes for levels could differ in both frames for the same level values.
+    private String _levelValue;
+    private Long _foldValue;
```

Comment:
Does this need to be a Long? The constructor takes a `long`. Shouldn't' int be enough?

Reply:
@michalkurka `Long` instead of `long` I wanted to use to be able to pass `null` for the case when we don't have fold column, but then forgot about it :). I used long since in general fold column could be defined by user and those values could be unexpectedly big ones( they might not be continuous sequence of numbers that starts from zero). Felt safer with long. But I agree, int will be more than enough in most cases. Thanks!